test/unit/lib/plugins/aws/deploy/lib/extendedValidate.test.js

```diff
@@ -193,5 +194,44 @@ describe('extendedValidate', () => {
         expect(awsDeploy.serverless.cli.log.firstCall.calledWithExactly(msg)).to.be.equal(true);
       });
     });
+
+    it("should not warn if function's timeout is greater than 30 and it's attached to APIGW, but it has [async] mode", () => {
+      stateFileMock.service.functions = {
+        first: {
+          timeout: 31,
+          package: {
+            artifact: 'artifact.zip',
+          },
+          events: [
+            {
+              http: {
+                async: true,
+              },
+            },
+          ],
+        },
+      };
+      awsDeploy.serverless.service.package.individually = true;
+      fileExistsSyncStub.returns(true);
+      readFileSyncStub.returns(stateFileMock);
+
+      return awsDeploy.extendedValidate().then(() => {
+        expect(awsDeploy.serverless.cli.log.firstCall).to.be.equal(null);
+      });
+    });
+
+    it("should not warn during the deployment if function's timeout is greater than 30 and it's attached to APIGW, but it has [async] mode", () => {
```

Comment:
Let's move this one to new top-level describe as suggested in  https://github.com/serverless/serverless/blob/master/test/README.md

Additionally, let's name the new describe with the path to this test file - `test/unit/lib/plugins/aws/deploy/lib/extendedValidate.test.js`